### PR TITLE
Created exchange-app frontend v3.1

### PR DESCRIPTION
**Pull request by jramcast**
The code is copied from v3. I only introduce a minor change to check an env variable.

### CODE REVIEW

File: exchange-application/frontend/v3.1/src/FetchUtils.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+async function fetchWithRetry (url, options=null, retries=3, msTimeout=3000) {
+    const timerPromise = createTimerPromise(msTimeout)
+    try {
+        const response = await Promise.race([fetch(url, options), timerPromise]);
+        if (!response.ok && process.env.REACT_APP_CATCH_HTTP_ERRORS) {
```
Comment (jramcast) [inline exchange-application/frontend/v3.1/src/FetchUtils.js:5]: Here's the only change. I added the env variable to the condition

File: exchange-application/frontend/v3.1/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Frontend Microservice v3.1
+
+**NOTE**: This version is almost identical to v3.0. It only adds control over the HTTP errors with an environment variable called **REACT_APP_CATCH_HTTP_ERRORS**.
```
Comment (jramcast) [inline exchange-application/frontend/v3.1/README.md:3]: Also added a mention to the env variable in the readme

File: exchange-application/frontend/v3.1/src/FetchUtils.js (hunk 1 above)
Comment (m-czernek) [reply exchange-application/frontend/v3.1/src/FetchUtils.js:5]: IMHO kind of a strange name for the variable. I'd prefer something like `REACT_APP_ENABLE_RETRY` or something similar. 

But it's a nitpick, adding +1.

Comment (jramcast) [reply exchange-application/frontend/v3.1/src/FetchUtils.js:5]: @m-czernek  What about `REACT_APP_ENABLE_RETRY_INVALID_REQUESTS`? REACT_APP_ENABLE_RETRY` is too general and it doesn't affect retries on timeouts or network failures.

Comment (m-czernek) [reply exchange-application/frontend/v3.1/src/FetchUtils.js:5]: Sure, that works better IMO.

Comment (jramcast) [reply exchange-application/frontend/v3.1/src/FetchUtils.js]: I changed the variable name to `REACT_APP_RETRY_INVALID_REQUESTS`. Thanks @m-czernek !

**Pull request merged**